Разработаны компоненты для макета чата.

evroka opened the pull request.

evroka committed:
Разработаны компоненты для макета чата.

vlukyanoff commented on my-chat/package.json:
```diff
@@ -0,0 +1,19 @@
+  "dependencies": {
+    "bootstrap": "^4.1.0",
+    "react": "^16.3.2",
+    "react-chat-elements": "^0.9.0",
```
Хитро ;-)

vlukyanoff commented on my-chat/src/components/Message.js:
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import logo from '../logo.png';
+import 'react-chat-elements/dist/main.css';
```
Из этой библиотеки только стили берутся? Если да, то стоит ли добавлять зависимость от библиотеки, когда то, что она предоставляет, легко реализовать самостоятельно?

evroka committed:
1. Основной компонент чата (Content) реализован как компонент-контейнер, который в state хранит массив сообщений.

2. Компонент Input реализован как контролируемый компонент.

3. Добавлена функция адаптирования аватарки под имя пользователя.

vlukyanoff commented on my-chat/src/components/Content.js:
```diff
@@ -1,12 +1,38 @@
+        this.setState(() => {
+            return {msgs: arr};
+          });
+    }
```
По этому методу несколько предложений:

1. push модифицирует исходный массив, а state напрямую модифицировать плохо, только с помощью setState
2. this.setState может принимать в качестве аргумента как объект с новым состоянием, так и функцию (подробнее здесь: https://reactjs.org/docs/react-component.html#setstate). Функция в первом аргументе получит текущий state.

В результате можно уменьшить код и сделать корректнее:
```
sendHandler(message) {
    this.setState((prevState) => {
        return {msgs: prevState.msgs.concat(message)}
    }
}
```

vlukyanoff commented on my-chat/src/components/Message.js:
```diff
@@ -1,21 +1,60 @@
+        n = n % 16;
+        let addr = n.toString().padStart(2, "0");
+        const resultPath = pathStart + str + '/' + addr + pathEnd;
+        return resultPath;
```
Не понял, как работает эта штука) На занятии расскажешь)

evroka committed:
Реализовано подключение к серверу с помощью Firebase.

evroka committed:
Добавлена обработка некорректных данных, приходящих из базы.

evroka committed:
Исправлен формат отображения даты отправки сообщения

evroka committed:
Добавлена возможность быстрого перехода наверх и вниз страницы.
